File: C#/Transformer/StringTransformer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using XmlBinder.Exception;

namespace XmlBinder.Transformer
{
	/// <summary>
	/// Transformer class to turn xml input to string and vice-versa.
	/// </summary>
    public class StringTransformer : XmlBinder.Interface.TransformerInterface
    {
		public static StringTransformer Instance = new StringTransformer();

        public object read(string _value)
        {
            return _value;
        }

        public string write(object _value)
        {
            if (_value is string)
            {
                return (string)_value;
            }
            else
            {
                throw new TransformerException("StringTransformer", "write", "the object of type " + _value.GetType().ToString() + " is not a string");
            }
        }
    }
}

File: Java/src/storm/xmlbinder/transformer/StringTransformer.java
package storm.xmlbinder.transformer;

/**
 * Class to transform String to String and vice-versa.
 * @author Storm <j-storm@hotmail.fr>
 *
 */
public class StringTransformer implements TransformerInterface 
{
	/*
	 * (non-Javadoc)
	 * @see storm.xmlbinder.transformer.TransformerInterface#read(java.lang.String)
	 */
	public Object read(String _value) 
	{
		return _value;
	}

	/*
	 * (non-Javadoc)
	 * @see storm.xmlbinder.transformer.TransformerInterface#write(java.lang.Object)
	 */
	public String write(Object _value) 
	{
		return (String)_value;
	}
}
